contracts/strategies/stkCvxFXS/stkCvxFxsVault.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "../../GenericVault.sol";
+
+struct EarnedData {
+    address token;
+    uint256 amount;
+}
+
+interface stkCvxFxsStrategy {
+    function harvest(address _caller, uint256 _minAmountOut)
+        external
+        returns (uint256 harvested);
+}
+
+interface ICvxFxsStaking {
+    function claimableRewards(address _account)
+        external
+        view
+        returns (EarnedData[] memory userRewards);
+}
+
+contract stkCvxFxsVault is GenericUnionVault {
+    bool public isHarvestPermissioned = false;
+    mapping(address => bool) public authorizedHarvesters;
+
+    constructor(address _token) GenericUnionVault(_token) {}
+
+    /// @notice Sets whether only whitelisted addresses can harvest
+    /// @param _status Whether or not harvests are permissioned
+    function setHarvestPermissions(bool _status) external onlyOwner {
+        isHarvestPermissioned = _status;
+    }
+
+    /// @notice Adds or remove an address from the harvesters' whitelist
+    /// @param _harvester address of the authorized harvester
+    /// @param _authorized Whether to add or remove harvester
+    function updateAuthorizedHarvesters(address _harvester, bool _authorized)
+        external
+        onlyOwner
+    {
+        authorizedHarvesters[_harvester] = _authorized;
+    }
+
+    /// @notice Claim rewards and swaps them to cvxFxs for restaking
+    /// @param _minAmountOut - min amount of cvxFxs to receive for harvest
+    /// @dev Can be called by whitelisted account or anyone against a cvxFxs incentive
+    /// @dev Harvest logic in the strategy/harvester contract
+    /// @dev Harvest can be called even if permissioned when last staker is
+    ///      withdrawing from the vault.
+    function harvest(uint256 _minAmountOut) public {
+        require(
```

Comment:
Potential alternative:

```
    error NotAuthorizedHarvester();

...

    function harvest(uint256 _minAmountOut) public {
        if (isHarvestPermissioned && !authorizedHarvesters[msg.sender]) revert NotAuthorizedHarvester();
```

I removed the `totalSupply() == 0` condition since it's only going to be truthy for a very brief period and would lead to unnecessary storage reads afterward.

Reply:
> I removed the totalSupply() == 0 condition since it's only going to be truthy for a very brief period and would lead to unnecessary storage reads afterward.

After taking a closer look, I can see why this condition is here (`GenericVault._withdraw` will call this method if the vault is emptied). Since the execution conditions are unlikely (if the pounder has at least 1 user or forgotten dust then the sequence will never be executed), it may not be efficient to have it.